Add data and privacy doc for WV2 (#2575)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/concepts/data-privacy.md

---
title: Data and privacy in WebView2
description: WebView2 data collection and practices, privacy, and the 'Diagnostic data' setting.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: webview
ms.date: 05/03/2023
---
# Data and privacy in WebView2

The WebView2 team tracks data to keep WebView2 secure and up to date, diagnose issues, and improve WebView2. This article is for WebView2 developers.
WebView2 developers consent to this data collection in the WebView2 Runtime Terms and Conditions License. Additionally, WebView2 follows the standards outlined in the [Microsoft Edge Privacy Whitepaper](../../privacy-whitepaper/index.md).

In general, WebView2 data collection goes through the same strict standards as Microsoft Edge data collection. The main telemetry sources are Chromium and Microsoft Edge telemetry infrastructure, Windows data reporting, and Watson (Microsoft Internal) infrastructure to collect crash dumps.

> Any WebView2 data collection complies with the laws, regulations, and best practices for privacy, and WebView2 has guardrails to protect against privacy issues.

Reviewer: ```suggestion
Any WebView2 data collection complies with the laws, regulations, and best practices for privacy, and WebView2 has mechanisms to ensure privacy.
```
improve wording at:  has guardrails to protect against privacy issues
*  guardrails = colloquialism; use more direct technical term.
*  try a consistently positive wording; "protecting against privacy" sounds opposite, like double-negative.

End users can control the data collection of WebView2 through the Windows **Diagnostic data** setting on Windows 10 and 11. As a developer, you are welcome to convey this to your users and invite them to use the Windows **Diagnostic data** setting to fit their preference.
WebView2 developers have no control over the Windows **Diagnostic data** setting. This is similar to the Windows APIs.

When the Windows **Diagnostic data** setting is turned **On**, WebView2-related data including API usage, SDK usage, and creation failure is collected. When this setting is **On**, browser diagnostic data is collected; however, personalization data is not collected. Only browser events that are relevant to WebView2 are collected. Regardless of the Windows **Diagnostic data** setting, WebView2 collects [Required data](https://support.microsoft.com/en-us/windows/diagnostics-feedback-and-privacy-in-windows-28808a2b-a31b-dd73-dcd3-4559a5199319) which is necessary to maintain performance and reliability.

> There are two APIs that exist to give developers more control over data collection: [Disable SmartScreen](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/IsSmartScreenRequired.md) and [Custom Crash Reports](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/DisableCrashReporting.md). See [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy) in _Overview of WebView2 features and APIs_.

Author: I can add the smartscreen ref and remove the doc, is it fine to keep the crash reporting md spec.

> There are two APIs that exist to give developers more control over data collection: [Disable SmartScreen](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/IsSmartScreenRequired.md) and [Custom Crash Reports](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/DisableCrashReporting.md). See [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy) in _Overview of WebView2 features and APIs_.

Reviewer: ```suggestion
The following APIs control data collection:

##### [.NET/C#](#tab/dotnetcsharp)

* `CoreWebView2Settings` Class:
   * [CoreWebView2Settings.IsReputationCheckingRequired Property](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2settings.isreputationcheckingrequired)
* `CoreWebView2EnvironmentOptions` Class:
   * [CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled Property](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2environmentoptions.iscustomcrashreportingenabled)
* `CoreWebView2Environment` Class:
   * [CoreWebView2Environment.FailureReportFolderPath Property](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2environment.failurereportfolderpath)

##### [WinRT/C#](#tab/winrtcsharp)

* `CoreWebView2Settings` Class:
   * [CoreWebView2Settings.IsReputationCheckingRequired Property](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings#isreputationcheckingrequired)
* `CoreWebView2EnvironmentOptions` Class:
   * [CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled Property](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environmentoptions#iscustomcrashreportingenabled)
* `CoreWebView2Environment` Class:
   * [CoreWebView2Environment.FailureReportFolderPath Property](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#failurereportfolderpath)

##### [Win32/C++](#tab/win32cpp)

* `ICoreWebView2Settings` interface:
   * [ICoreWebView2Settings::get_IsReputationCheckingRequired](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2settings8#get_isreputationcheckingrequired)
   * [ICoreWebView2Settings::put_IsReputationCheckingRequired](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2settings8#put_isreputationcheckingrequired)
* `ICoreWebView2EnvironmentOptions3` interface:
   * [ICoreWebView2EnvironmentOptions3::get_IsCustomCrashReportingEnabled](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2environmentoptions3#get_iscustomcrashreportingenabled)
* `ICoreWebView2Environment11` interface:
   * [ICoreWebView2Environment11::get_FailureReportFolderPath](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2environment11#get_failurereportfolderpath)

---

See also:
* [Spec: Disable SmartScreen](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/IsSmartScreenRequired.md) - about `IsReputationCheckingRequired`.
* [Spec: Custom Crash Reports](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/DisableCrashReporting.md) - about `IsCustomCrashReportingEnabled` and `FailureReportFolderPath`.
* [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy) in _Overview of WebView2 features and APIs_.
```

<!-- ====================================================================== -->
## SmartScreen

> Microsoft Defender SmartScreen ("SmartScreen") is enabled by default. If you don't disable SmartScreen, you must provide notice to all users that your software includes Microsoft Defender SmartScreen and collects and sends the end user's information to Microsoft as disclosed in [Microsoft Privacy Statement](https://aka.ms/privacy) and the [Microsoft Edge Privacy Whitepaper](../../privacy-whitepaper/index.md#smartscreen). All other services in `edge://settings/privacy` should be turned off.

Reviewer: ```suggestion
Microsoft Defender SmartScreen ("SmartScreen") is enabled by default. If you don't disable SmartScreen, you must provide notice to all users that your software includes Microsoft Defender SmartScreen and collects and sends the user's information to Microsoft as disclosed in [Microsoft Privacy Statement](https://aka.ms/privacy) and the [Microsoft Edge Privacy Whitepaper](../../privacy-whitepaper/index.md#smartscreen). All other services in `edge://settings/privacy` should be turned off.
```
consistently "user" or "end user"

> See [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy) in _Overview of WebView2 features and APIs_.

Reviewer: ```suggestion
The Managing SmartScreen API includes the `IsReputationCheckingRequired` property, which controls whether SmartScreen is enabled.  See [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy) in _Overview of WebView2 features and APIs_.
```

> The managing SmartScreen API includes the `IsReputationCheckingRequired` property, which controls whether SmartScreen is enabled.  See [Security and privacy](../concepts/overview-features-apis.md#smartscreen) in _Overview of WebView2 features and APIs_.

Reviewer: ```suggestion
The managing SmartScreen API includes the `IsReputationCheckingRequired` property, which controls whether SmartScreen is enabled.  See [SmartScreen](../concepts/overview-features-apis.md#smartscreen) in _Overview of WebView2 features and APIs_.
```
update link text
note that the See Also section below has a broader link; it links to the parent h4 section

---
title: Data and privacy in WebView2
description: WebView2 data collection and practices, privacy, and the 'Diagnostic data' setting.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: webview

> ms.date: 05/03/2023

Reviewer: ```suggestion
ms.date: 05/16/2023
```
motivation: see if doc build reports a false 404

Reviewer: this thread can be closed, per latest https://github.com/MicrosoftDocs/edge-developer/pull/2575/files

Users can control the data collection of WebView2 through the Windows **Diagnostic data** setting on Windows 10 and 11. As a developer, you are welcome to convey this to your users and invite them to use the Windows **Diagnostic data** setting to fit their preference.
WebView2 developers have no control over the Windows **Diagnostic data** setting. This is similar to the Windows APIs.

When the Windows **Diagnostic data** setting is turned **On**, WebView2-related data including API usage, SDK usage, and creation failure is collected. When this setting is **On**, browser diagnostic data is collected; however, personalization data is not collected. Only browser events that are relevant to WebView2 are collected. Regardless of the Windows **Diagnostic data** setting, WebView2 collects required data that's necessary to maintain performance and reliability; see [Diagnostics, feedback, and privacy in Windows](https://support.microsoft.com/en-us/windows/diagnostics-feedback-and-privacy-in-windows-28808a2b-a31b-dd73-dcd3-4559a5199319).

> The following APIs control data collection:
> ##### [.NET/C#](#tab/dotnetcsharp)
> * `CoreWebView2Settings` Class:
>    * [CoreWebView2Settings.IsReputationCheckingRequired Property](/dotnet/api/microsoft.web.webview2.core.corewebview2settings.isreputationcheckingrequired)
> * `CoreWebView2EnvironmentOptions` Class:
>    * [CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled Property](/dotnet/api/microsoft.web.webview2.core.corewebview2environmentoptions.iscustomcrashreportingenabled)
> * `CoreWebView2Environment` Class:
>    * [CoreWebView2Environment.FailureReportFolderPath Property](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.failurereportfolderpath)
> ##### [WinRT/C#](#tab/winrtcsharp)
> * `CoreWebView2Settings` Class:
>    * [CoreWebView2Settings.IsReputationCheckingRequired Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings#isreputationcheckingrequired)
> * `CoreWebView2EnvironmentOptions` Class:
>    * [CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environmentoptions#iscustomcrashreportingenabled)
> * `CoreWebView2Environment` Class:
>    * [CoreWebView2Environment.FailureReportFolderPath Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#failurereportfolderpath)
> ##### [Win32/C++](#tab/win32cpp)
> * `ICoreWebView2Settings` interface:
>    * [ICoreWebView2Settings::get_IsReputationCheckingRequired](/microsoft-edge/webview2/reference/win32/icorewebview2settings8#get_isreputationcheckingrequired)
>    * [ICoreWebView2Settings::put_IsReputationCheckingRequired](/microsoft-edge/webview2/reference/win32/icorewebview2settings8#put_isreputationcheckingrequired)
> * `ICoreWebView2EnvironmentOptions3` interface:
>    * [ICoreWebView2EnvironmentOptions3::get_IsCustomCrashReportingEnabled](/microsoft-edge/webview2/reference/win32/icorewebview2environmentoptions3#get_iscustomcrashreportingenabled)
> * `ICoreWebView2Environment11` interface:
>    * [ICoreWebView2Environment11::get_FailureReportFolderPath](/microsoft-edge/webview2/reference/win32/icorewebview2environment11#get_failurereportfolderpath)

Reviewer: ```suggestion
The following APIs control data collection:

##### [.NET/C#](#tab/dotnetcsharp)

* `CoreWebView2Settings` Class:
   * [CoreWebView2Settings.IsReputationCheckingRequired Property](/dotnet/api/microsoft.web.webview2.core.corewebview2settings.isreputationcheckingrequired)
* `CoreWebView2EnvironmentOptions` Class:
   * [CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled Property](/dotnet/api/microsoft.web.webview2.core.corewebview2environmentoptions.iscustomcrashreportingenabled)
* `CoreWebView2Environment` Class:
   * [CoreWebView2Environment.FailureReportFolderPath Property](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.failurereportfolderpath)

##### [WinRT/C#](#tab/winrtcsharp)

* `CoreWebView2Settings` Class:
   * [CoreWebView2Settings.IsReputationCheckingRequired Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2settings#isreputationcheckingrequired)
* `CoreWebView2EnvironmentOptions` Class:
   * [CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environmentoptions#iscustomcrashreportingenabled)
* `CoreWebView2Environment` Class:
   * [CoreWebView2Environment.FailureReportFolderPath Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#failurereportfolderpath)

##### [Win32/C++](#tab/win32cpp)

* `ICoreWebView2Settings` interface:
   * [ICoreWebView2Settings::get_IsReputationCheckingRequired](/microsoft-edge/webview2/reference/win32/icorewebview2settings8#get_isreputationcheckingrequired)
   * [ICoreWebView2Settings::put_IsReputationCheckingRequired](/microsoft-edge/webview2/reference/win32/icorewebview2settings8#put_isreputationcheckingrequired)
* `ICoreWebView2EnvironmentOptions3` interface:
   * [ICoreWebView2EnvironmentOptions3::get_IsCustomCrashReportingEnabled](/microsoft-edge/webview2/reference/win32/icorewebview2environmentoptions3#get_iscustomcrashreportingenabled)
* `ICoreWebView2Environment11` interface:
   * [ICoreWebView2Environment11::get_FailureReportFolderPath](/microsoft-edge/webview2/reference/win32/icorewebview2environment11#get_failurereportfolderpath)

---
```

Reviewer: this thread can be closed, per latest https://github.com/MicrosoftDocs/edge-developer/pull/2575/files

The managing SmartScreen API includes the `IsReputationCheckingRequired` property, which controls whether SmartScreen is enabled.  See [Smartscreen](../concepts/overview-features-apis.md#smartscreen) in _Overview of WebView2 features and APIs_.

<!-- ====================================================================== -->
## See also

* [Spec: Disable SmartScreen](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/IsSmartScreenRequired.md) - about `IsReputationCheckingRequired`.
* [Spec: Custom Crash Reports](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/specs/DisableCrashReporting.md) - about `IsCustomCrashReportingEnabled` and `FailureReportFolderPath`.

> * [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy) in _Overview of WebView2 features and APIs_.

Reviewer: this thread can be closed

Users can control the data collection of WebView2 through the Windows **Diagnostic data** setting on Windows 10 and 11. As a developer, you are welcome to convey this to your users and invite them to use the Windows **Diagnostic data** setting to fit their preference.
WebView2 developers have no control over the Windows **Diagnostic data** setting. This is similar to the Windows APIs.

When the Windows **Diagnostic data** setting is turned **On**, WebView2-related data including API usage, SDK usage, and creation failure is collected. When this setting is **On**, browser diagnostic data is collected; however, personalization data is not collected. Only browser events that are relevant to WebView2 are collected. Regardless of the Windows **Diagnostic data** setting, WebView2 collects required data that's necessary to maintain performance and reliability; see [Diagnostics, feedback, and privacy in Windows](https://support.microsoft.com/en-us/windows/diagnostics-feedback-and-privacy-in-windows-28808a2b-a31b-dd73-dcd3-4559a5199319).

> The following APIs control data collection:

Reviewer 2: ```suggestion
While developers do not have control over the overall diagnostic data collection, WebView2 offers the following APIs to control the behavior of specific features that also generates data: 
```

* `ICoreWebView2EnvironmentOptions3` interface:
   * [ICoreWebView2EnvironmentOptions3::get_IsCustomCrashReportingEnabled](/microsoft-edge/webview2/reference/win32/icorewebview2environmentoptions3#get_iscustomcrashreportingenabled)
* `ICoreWebView2Environment11` interface:
   * [ICoreWebView2Environment11::get_FailureReportFolderPath](/microsoft-edge/webview2/reference/win32/icorewebview2environment11#get_failurereportfolderpath)

---

<!-- ====================================================================== -->
## SmartScreen

Microsoft Defender SmartScreen ("SmartScreen") is enabled by default. If you don't disable SmartScreen, you must provide notice to all users that your software includes Microsoft Defender SmartScreen and collects and sends the user's information to Microsoft as disclosed in [Microsoft Privacy Statement](https://aka.ms/privacy) and the [Microsoft Edge Privacy Whitepaper](../../privacy-whitepaper/index.md#smartscreen). Generally, all other services in `edge://settings/privacy` are turned off, for WebView2.

> The managing SmartScreen API includes the `IsReputationCheckingRequired` property, which controls whether SmartScreen is enabled.  See [Smartscreen](../concepts/overview-features-apis.md#smartscreen) in _Overview of WebView2 features and APIs_.

Reviewer: ```suggestion
The managing SmartScreen API includes the `IsReputationCheckingRequired` property, which controls whether SmartScreen is enabled.  See [Security and privacy](../concepts/overview-features-apis.md#security-and-privacy)<!--todo in next PR: link specifically to #smartscreen instead--> in _Overview of WebView2 features and APIs_.
```
